modules/opengl/src/shader/shader.cpp

```diff
@@ -97,11 +112,11 @@ Shader::Shader(const Shader &rhs) : id_{glCreateProgram()}, warningLevel_{rhs.wa
     ShaderManager::getPtr()->registerShader(this);
 }
 
-Shader::Shader(Shader &&rhs) : id_{rhs.id_}, ready_(rhs.ready_), warningLevel_{rhs.warningLevel_} {
+Shader::Shader(Shader &&rhs)
+    : prog_{rhs.prog_.id}, ready_(rhs.ready_), warningLevel_{rhs.warningLevel_} {
```

Comment:
This should be done by the wapper class

Reply:
Not possible since we need to deregister the rhs shader as far as I understood. This applies in a couple of situations where it is not possible to do something without knowing about the shader (assignment). I therefore tried to keep the ScopedShaderProgram as light-weight as possible